Mediate 프로젝트

wonkyoungChoi opened the pull request:
대학교 검색 및 선택 기능 완료
학년 선택 기능 완료(Spinner 사용)
학과 정보 받아오기 기능 구현중

wonkyoungChoi committed:
Mediate 프로젝트
대학교 검색 및 선택 기능 완료
학년 선택 기능 완료(Spinner 사용)
학과 정보 받아오기 기능 구현중

naglwo commented on app/src/main/java/com/wk/mediate/repository/SearchSchoolRepository.kt:
```diff
@@ -48,4 +50,43 @@ class SearchSchoolRepository {
+                val num: Int
+                if (list != null) {
+                    try {
+                        num = if (list.dataSearch.content[0].totalCount > 20) 5
```
int 5를 그대로 변수에 넣는거보다는 변수로 빼서 처리하는 게 보기 좋을 것 같아ㅏ!!

wonkyoungChoi replied:
알았어 이부분도 수정해볼게

naglwo commented on app/src/main/java/com/wk/mediate/repository/SearchSchoolRepository.kt:
```diff
@@ -48,4 +50,43 @@ class SearchSchoolRepository {
+                        num = if (list.dataSearch.content[0].totalCount > 20) 5
+                        else list.dataSearch.content[0].totalCount - 1
+                        Log.d("NUM", num.toString())
+                        for (i in 0..num) {
```
num.forEach { it -> items.add(it) } 이런식으로 forEach문 써서 바꿔보면 더 간결해질듯?

wonkyoungChoi replied:
오키오키 수정해볼게!

naglwo commented on app/src/main/java/com/wk/mediate/ui/Register/Search/SearchSchoolActivity.kt:
```diff
@@ -6,53 +6,135 @@ import android.text.Editable
     private lateinit var items: ArrayList<String>
+//    private val name : String? = intent.getStringExtra("name")
+//    private val phoneNum : String? = intent.getStringExtra("phoneNum")
+    private val type : String = "tutor"
```
type이 변하지 않는 string 값이면 companion object 상수로 뺴서 사용하는 것도 좋을 것 같아!

wonkyoungChoi replied:
이 값은 테스트해보려고 잠깐 넣어둔거라 나중에 바뀔거 같아

SimonMJC commented on app/src/main/java/com/wk/mediate/ui/Register/Search/SearchUniversityDTO.kt:
```diff
@@ -0,0 +1,15 @@
+        val content: List<Content>
+)
+
+data class Content(
```
데이터 클래스들은 nullable(?) 로처리하는 게 안전할 듯!! 데이터에 널이 들어올 가능성이 충분할 수도 있고, model에서 받아올 때나 viewmodel에서 데이터 가공할 때 null 체크 하는 방향으로?
data class DataTransfer(val parent: String?, val child: String?, val child2: String?)

wonkyoungChoi replied:
확인하고 수정도 완료했어!

wonkyoungChoi committed:
Mediate 프로젝트
Pull Request 수정 필요한 부분 수정 완료
학과 정보 받아오기 기능 구현중(검색하여 얻는 값이 정상적이지 않아 새로운 방법을 찾아야함)

wonkyoungChoi committed:
Mediate 프로젝트
학교//학년 검색 기능 및 UI 구현 완료
학과 정보 받아오기 기능 구현중(검색하여 얻는 값이 정상적이지 않아 새로운 방법을 찾아야함)

wonkyoungChoi commented:
머지하고 푸쉬를 다시해야하는데 푸쉬하고 머지를 해서 방금 수정한 내용들은 바로 머지가 된거같아..!
